conan/tools/cmake/cmakedeps2/cmakedeps.py

```diff
@@ -243,14 +248,15 @@ def generate(self):
 
         # CMAKE_PROGRAM_PATH | CMAKE_LIBRARY_PATH | CMAKE_INCLUDE_PATH
         cmake_program_path = self._get_cmake_paths([(req, dep) for req, dep in all_reqs if req.direct], "bindirs")
-        cmake_library_path = self._get_cmake_paths(list(host_req.items()) + list(test_req.items()), "libdirs")
-        cmake_include_path = self._get_cmake_paths(list(host_req.items()) + list(test_req.items()), "includedirs")
-
+        cmake_library_path = self._get_cmake_paths(host_test_reqs, "libdirs")
+        cmake_include_path = self._get_cmake_paths(host_test_reqs, "includedirs")
+        cmake_framework_path = self._get_cmake_paths(host_test_reqs, "frameworkdirs")
```

Comment:
Lets make sure to document this as necessary, because linking can work without it, but this ``CMAKE_FRAMEWORK_PATH`` would fail without it